docs/architecture/adr-011-light-client-crates-extraction.md

```diff
@@ -0,0 +1,292 @@
+# ADR 011: Light client crates extraction
+
+## Changelog
+
+* 2022-07-29: Initial draft
+
+## Context
+
+The current design of the `ibc` (modules) crate precludes light client implementations from being hosted outside of the
+ibc-rs repository. This is primarily due to circular dependencies and tight coupling between the light client
+implementations and the `ibc` crate.
+
+For e.g., the Tendermint light client implementation (rightly) depends on the `ibc` crate for the trait definitions
+(such as `ClientState`) that a light client is expected to implement, but the `ibc` crate also depends on the Tendermint
+light client's `ClientState` impl to define its `AnyClientState` enum:
+
+```rust
+pub enum AnyClientState {
+    Tendermint(client_state::ClientState),
+
+    #[cfg(any(test, feature = "mocks"))]
+    Mock(MockClientState),
+}
+```
+
+See [ADR003 - Dealing with chain-specific datatypes](https://github.com/informalsystems/ibc-rs/blob/master/docs/architecture/adr-003-handler-implementation.md#dealing-with-chain-specific-datatypes)
```

Comment:
Adding a summary of the rationale behind the initial decision for the `Any*` enum here so that folks reading this ADR don't have to click away to a different doc (unless they want the additional context) would be helpful here.

Reply:
Good point, done! https://github.com/informalsystems/ibc-rs/pull/2356/commits/f4a7747255c310b92b2ca4b878cd88c7875efd11